Azure/azure-cli
# [App Service] `az functionapp create`: Add new parameter `--workspace` to support creating workspace-based app insights components

**Related command**
`az functionapp create`

**Description**
Currently, during the function app creation flow, we create a classic app insights component. However, this will no longer be supported in February 2024 so we have to create workspace-based app insights components instead. The Azure Portal already does this so we need to implement the same behavior in the Azure CLI.

When creating a function app in the Azure Portal, a resource group named `DefaultResourceGroup-<RegionCode>` is created and then a workspace named `DefaultWorkspace-{SubscriptionId>-<RegionCode>` is created inside that default resource group. Then that default workspace is attached to the application insights component that gets created.

Relevant documentation: https://learn.microsoft.com/en-us/azure/azure-monitor/app/convert-classic-resource

**Testing Guide**
1. Create function app with `az functionapp create` command. I have also added a new param `--workspace` that accepts the name of an existing log analytics workspace resource to be used for the application insights component that will be created.
2. Check that the app insights component has a workspace resource associated with it. This can be confirmed either through the Azure Portal or by running `az monitor app-insights component show` and looking for `workspaceResourceId`

---

This checklist is used to make sure that common guidelines for a pull request are followed.
